tests/unit/utils/validate-dns-test.js

```diff
@@ -0,0 +1,29 @@
+import { module, test } from 'qunit';
+import {validateHostname} from "ui/utils/validate-dns";
+
+module('Unit | Utils | validate-dns');
+
+//test case
+let hostnames = [
+    {str:"aA0",result:true},
+    //chec for dot
+    {str:".aA0",result:false},
+    {str:"aA0.",result:true},
+    {str:"a.A0",result:true},
+    {str:"a..A0",result:false},
+    //check for dash
+    {str:"a-A0",result:true},
+    {str:"-aA0",result:false},
+    {str:"aA0-",result:false},
+    {str:"a--A0",result:false},
+    //check for specifical character
+    {str:"aA0/",result:false},
+    {str:"a/A0",result:false},
+    {str:"/aA0",result:false},
```

Comment:
Add tests for empty string, a label is too long, and the entire hostname is too long?